apps/src/lib/kits/maker/boards/microBit/LightSensor.js

```diff
@@ -10,8 +15,19 @@ export default class LightSensor extends EventEmitter {
       rangeMax: 255,
       currentReading: 0
     };
+    // Track 3 seconds of historical data over which we can average sensor readings
+    this.buffer = new Float32Array(MAX_SENSOR_BUFFER / SAMPLE_RATE);
+    this.bufferIndex = 0;
+
     this.board = board;
     this.board.mb.addFirmataUpdateListener(() => {
+      //Record current reading to keep finite historical buffer
+      this.buffer[this.bufferIndex] = this.board.mb.analogChannel[
```

Comment:
I [noticed](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Float32Array) that this data type is stored in "platform byte order".  Does the assignment from this board interface into our buffer value always work, given this?

Reply:
Hi there! I don't know what this means. If you know, I'd love to chat about it, otherwise I'll do some research and see what I can find.

Reply:
Discussed offline. Decided this implementation works.